python/cudf/cudf/core/dataframe.py

```diff
@@ -5749,8 +5750,11 @@ def to_arrow(self, preserve_index=None) -> pa.Table:
             preserve_index=preserve_index,
             types=out.schema.types,
         )
+        md_dict = json.loads(metadata[b"pandas"])
 
-        return out.replace_schema_metadata(metadata)
+        cudf.utils.ioutils._post_process_dtypes(self, md_dict)
```

Comment:
nit on adding a comment that we're modifying the `metadata_dict` inplace here to make sure the numpy_types (and whatever else) are now typed in a fashion that pandas / pyarrow likes?

Reply:
Done.